fanficfare/adapters/adapter_deviantartcom.py

```diff
@@ -0,0 +1,187 @@
+            'challenge': soup.find('input', {'name': 'challenge'})['value'],
+        }
+
+        if self.password:
+            params['username'] = self.username
+            params['password'] = self.password
+        else:
+            params['username'] = self.getConfig('username')
+            params['password'] = self.getConfig('password')
+
+        loginUrl = 'https://' + self.getSiteDomain() + '/_sisu/do/signin'
+        logger.debug('Will now login to deviantARt as (%s)' % params['username'])
+
+        result = self.post_request(loginUrl, params, usecache=False)
+
+        if 'Log In | DeviantArt' in result:
+            logger.error('Failed to login to deviantArt as %s' % params['username'])
+            raise exceptions.FailedToLogin('https://www.deviantart.com', params['username'])
+        else:
+            return True
+
+    def requiresLogin(self, data):
+        return '</a> has limited the viewing of this artwork to members of the DeviantArt community only' in data
+
+    def isWatchersOnly(self, data):
+        return '<span>Watchers-Only Deviation</span>' in data
+
+    def requiresMatureContentEnabled(self, data):
+        return (
+            '>This content is intended for mature audiences<' in data
+            or '>This deviation is intended for mature audiences<' in data
+        )
+
+    def extractChapterUrlsAndMetadata(self):
+        isLoggedIn = False
+        logger.debug('URL: %s', self.url)
+
+        data = self.get_request(self.url)
+
+        soup = self.make_soup(data)
+
+        if self.requiresLogin(data):
+            if self.performLogin(self.url):
+                isLoggedIn = True
+                data = self.get_request(self.url, usecache=False)
+                soup = self.make_soup(data)
+
+        if self.isWatchersOnly(data):
+            raise exceptions.FailedToDownload(
+                'Deviation is only available for watchers.' +
+                'You must watch this author before you can download it.'
+            )
+
+        if self.requiresMatureContentEnabled(data):
+            # as far as I can tell deviantArt has no way to show mature
+            # content that doesn't involve logging in or using JavaScript
+            if not isLoggedIn:
+                self.performLogin(self.url)
+                isLoggedIn = True
+                data = self.get_request(self.url, usecache=False)
+                soup = self.make_soup(data)
+                if self.requiresMatureContentEnabled(data):
+                    raise exceptions.FailedToDownload(
+                        'Deviation is set as mature, you must go into your account ' +
+                        'and enable showing of mature content.'
+                    )
+
+        title = soup.select_one('h1').get_text()
+        self.story.setMetadata('title', stripHTML(title))
+
+        ## dA has no concept of status
+        # self.story.setMetadata('status', 'Completed')
```

Comment:
I wasn't sure of the status to use here, since deviantArt doesn't have that concept.